executor/main.go

```diff
@@ -145,10 +146,24 @@ func runGrpcServer(lis net.Listener, logger logr.Logger, predictor *v1.Predictor
 	}
 	err = grpcServer.Serve(lis)
 	if err != nil {
-		log.Errorf("Grpc server error: %v", err)
+		logger.Error(err, "gRPC server error")
 	}
 }
 
+func setupLogger(debug bool) {
```

Comment:
If if `setupLogger` is setting the value of `debug` variable, wouldn't it be better to do it through the return value?

Reply:
It's true that right now that method is not paying much attention to the `debug` flag. This is linked to the conversation above. It may make sense to separate the log level and the debug / production flag of the logs.